standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/columnstats/aggr/DateColumnStatsAggregatorTest.java

```diff
@@ -0,0 +1,279 @@
+    ColumnStatisticsData data2 = new ColStatsBuilder<>(Date.class).numNulls(2).numDVs(3).build();
+
+    List<ColStatsObjWithSourceInfo> statsList = Arrays.asList(createStatsWithInfo(data1, TABLE, COL, partitions.get(0)),
+        createStatsWithInfo(data2, TABLE, COL, partitions.get(1)));
+
+    DateColumnStatsAggregator aggregator = new DateColumnStatsAggregator();
+
+    ColumnStatisticsObj computedStatsObj = aggregator.aggregate(statsList, partitions, true);
+    ColumnStatisticsData expectedStats = new ColStatsBuilder<>(Date.class).numNulls(3).numDVs(3)
+        .low(DATE_1).high(DATE_2).hll(values1).build();
+    Assert.assertEquals(expectedStats, computedStatsObj.getStatsData());
+
+    aggregator.useDensityFunctionForNDVEstimation = true;
+    computedStatsObj = aggregator.aggregate(statsList, partitions, true);
+    expectedStats = new ColStatsBuilder<>(Date.class).numNulls(3).numDVs(4)
+        .low(DATE_1).high(DATE_2).hll(values1).build();
+    Assert.assertEquals(expectedStats, computedStatsObj.getStatsData());
+
+    aggregator.useDensityFunctionForNDVEstimation = false;
+    aggregator.ndvTuner = 1;
+    computedStatsObj = aggregator.aggregate(statsList, partitions, true);
+    expectedStats = new ColStatsBuilder<>(Date.class).numNulls(3).numDVs(5)
+        .low(DATE_1).high(DATE_2).hll(values1).build();
+    Assert.assertEquals(expectedStats, computedStatsObj.getStatsData());
+  }
+
+  @Test
+  public void testAggregateMultiStatsWhenAllAvailable() throws MetaException {
+    List<String> partitions = Arrays.asList("part1", "part2", "part3");
+
+    long[] values1 = { DATE_1.getDaysSinceEpoch(), DATE_2.getDaysSinceEpoch(), DATE_3.getDaysSinceEpoch() };
+    ColumnStatisticsData data1 = new ColStatsBuilder<>(Date.class).numNulls(1).numDVs(3)
+        .low(DATE_1).high(DATE_3).hll(values1).build();
+
+    long[] values2 = { DATE_3.getDaysSinceEpoch(), DATE_4.getDaysSinceEpoch(), DATE_5.getDaysSinceEpoch() };
+    ColumnStatisticsData data2 = new ColStatsBuilder<>(Date.class).numNulls(2).numDVs(3)
+        .low(DATE_3).high(DATE_5).hll(values2).build();
+
+    long[] values3 = { DATE_6.getDaysSinceEpoch(), DATE_7.getDaysSinceEpoch() };
+    ColumnStatisticsData data3 = new ColStatsBuilder<>(Date.class).numNulls(3).numDVs(2)
+        .low(DATE_6).high(DATE_7).hll(values3).build();
+
+    List<ColStatsObjWithSourceInfo> statsList = Arrays.asList(createStatsWithInfo(data1, TABLE, COL, partitions.get(0)),
+        createStatsWithInfo(data2, TABLE, COL, partitions.get(1)), createStatsWithInfo(data3, TABLE, COL, partitions.get(2)));
+
+    DateColumnStatsAggregator aggregator = new DateColumnStatsAggregator();
+    ColumnStatisticsObj computedStatsObj = aggregator.aggregate(statsList, partitions, true);
+
+    // the aggregation does not update hll, only numNDVs is, it keeps the first hll
+    ColumnStatisticsData expectedStats = new ColStatsBuilder<>(Date.class).numNulls(6).numDVs(7)
```

Comment:
This test is to show how the `numDV` is not computed by using the `numDV` from the different partitions (which is not composable, since we can't estimate if there is an intersection or not), but using the bitvector.

If you look at the number of distinct values it's 7 and not 8 because `DATE_3` appears twice, hence it must be counted only once.

Reply:
Got it. Maybe this explanation deserves to be reflected on the test name or in the comment. While I was going over the tests I was mostly comparing numDVs from data with expected numDVs. It took me a while to notice that it is taken from the hll of the data.

Reply:
It's a pretty crucial point, I agree it deservers a better comment in the assert part, I will improve it.